### ISSUE TITLE
Says "No companies found for this problem" even for a famous problem

### ISSUE THREAD

**Issue by phoeniXharsh** [codedecks-in/Big-Omega-Extension#43]
As far as I can remember, this problem wasn't present before so it must be a bug. 
![image](https://user-images.githubusercontent.com/60690724/231845534-3032cbf3-d505-4dc4-86ea-748f468a1478.png)

An interesting thing is that, now it unblurs the company tags in the "Companies" section. So we can see the companies for 0-6 months but the default extension dashboard for listing all the companies is missing. 
![image](https://user-images.githubusercontent.com/60690724/231846019-fbb88a69-919e-4c9e-9e18-75a9daeeb0f4.png)

**Comment by Arnab7456** [comment]
365. Water and Jug Problem 
this question also asked in Vmware

**Comment by GouravRusiya30** [comment]
Thanks, we will update the list.

**Comment by omkarajagunde** [comment]
@Rishabh-Pahwa yes we did add unblur styling in recent published version 1.1.0

**Comment by GouravRusiya30** [comment]
Does anyone wants to pick this up and update the list ?
We are happy to take the PRs but please do add the source of the updated tagged companies list.

**Comment by Rishabh-Pahwa** [comment]
Do we need to add the companies and problems to the json file?

**Comment by RohanDebroy** [comment]
Moving the company data to a database rather than storing it in a JSON file would be better. That way it would become easier for people to contribute and keep the list up-to-date.

**Comment by GouravRusiya30** [comment]
@RohanDebroy
We have already discussed about storing the company data in a DB but there are tradeoffs for it. Some of them includes :
- Increase in cost
- Maintenance of DB
- Handling the latency etc

Rather than we can push this data in another Github repo and store there as a json. We can utilise the Github APIs to fetch the data. Pros with this approach:
- Easy maintenance
- Easily updatable
- Separated from the extension package
- Modular approach

Please let me know if anyone is interested to pick this Task. We will be happy to discuss & help further.

Thanks !!!

**Comment by Rishabh-Pahwa** [comment]
I would love to work on this

**Comment by GouravRusiya30** [comment]
Great !!!
Please start working on this, I have assigned this issue to you. Let me know if you want to discuss on this.

**Comment by Rishabh-Pahwa** [comment]
Yes, I would like to discuss a few points.

**Comment by GouravRusiya30** [comment]
@Rishabh-Pahwa Can you write up a draft doc with the implementation approach and share us via email - `manage.bigomega@gmail.com`

After which we can have a google meet schedule to discuss further.

**Comment by GouravRusiya30** [comment]
@Rishabh-Pahwa Are you working on this?

**Comment by Rishabh-Pahwa** [comment]
Yes, will mail you the write up doc soon

**Comment by DarshanDixit05** [comment]
Hi @GouravRusiya30 👋 , I want to work on this issue. Can you please assign it to me?

**Comment by WOLFIEEEE** [comment]
Hey @GouravRusiya30  @Rishabh-Pahwa  

Rather than we can push this data into another Github repo and store it there as a JSON. We can utilize the GitHub APIs to fetch the data. Pros of this approach:

Easy maintenance
Easily updatable
Separated from the extension package
Modular approach

Seems a good approach However, as I delved deeper into implementing this solution ( Just Playing around), I couldn't help but ponder the potential drawbacks. The main concern revolves around GitHub's API rate limiting. In its current state, unauthorized requests are limited to 60 per hour, while authorized ones (using personal access tokens) are capped at 5000 requests. Even if we implement caching strategies to minimize API calls, the rate limits may still hinder the smooth functioning of our extension. This becomes especially crucial since we aim to promptly update the data across all systems once a change is made.

I just wanted to share this insight with you all and express my concern. If any of you have come across a potential workaround or solution for this limitation, Coool !! looking forward to the update.

**Comment by Rishabh-Pahwa** [comment]
@GouravRusiya30 Have you checked my email? When can we have the Google meet?

**Comment by GouravRusiya30** [comment]
@Rishabh-Pahwa yes, I have checked the doc but I was expecting a more detailed one. Please deep-dive more on the implementation part of API invocation.
Leave apart the README and UI changes. We are more interested in the Backend implementation right now.

- README is secondary
- UI we won't need ❌

@WOLFIEEEE very good point & I have already discussed this caching implementation with @omkarajagunde and below are the points that we agreed upon

- Invoke Github API once/twice per day & let it be delayed for a day for now
